impl/client/src/main/javascript/web/dojo/pentaho/common/overrides/dijit/_PaletteMixin.js

```diff
@@ -0,0 +1,357 @@
+		//		dyeClass should implement the dijit/_PaletteMixin.__Dye interface.
+		dyeClass: null,
+
+		_dyeFactory: function(value /*===== , row, col, title =====*/){
+			// summary:
+			//		Return instance of dijit.Dye for specified cell of palette
+			// tags:
+			//		extension
+
+			// Remove string support for 2.0
+			var dyeClassObj = typeof this.dyeClass == "string" ? lang.getObject(this.dyeClass) : this.dyeClass;
+			return new dyeClassObj(value);
+		},
+
+		_preparePalette: function(choices, titles){
+			// summary:
+			//		Subclass must call _preparePalette() from postCreate(), passing in the tooltip
+			//		for each cell
+			// choices: String[][]
+			//		id's for each cell of the palette, used to create Dye JS object for each cell
+			// titles: String[]
+			//		Localized tooltip for each cell
+
+			this._cells = [];
+			var url = this._blankGif;
+
+			this.own(on(this.gridNode, a11yclick, lang.hitch(this, "_onCellClick")));
+
+			for(var row = 0; row < choices.length; row++){
+				var rowNode = domConstruct.create("tr", {tabIndex: "-1", role: "row"}, this.gridNode);
+				for(var col = 0; col < choices[row].length; col++){
+					var value = choices[row][col];
+					if(value){
+						var cellObject = this._dyeFactory(value, row, col, titles[value]);
+
+						var cellNode = domConstruct.create("td", {
+							"class": this.cellClass,
+							tabIndex: "-1",
+							title: titles[value],
+							role: "gridcell"
+						}, rowNode);
+
+						// prepare cell inner structure
+						cellObject.fillCell(cellNode, url);
+
+						cellNode.idx = this._cells.length;
+
+						// save cell info into _cells
+						this._cells.push({node: cellNode, dye: cellObject});
+					}
+				}
+			}
+			this._xDim = choices[0].length;
+			this._yDim = choices.length;
+
+			// Now set all events
+			// The palette itself is navigated to with the tab key on the keyboard
+			// Keyboard navigation within the Palette is with the arrow keys
+			// Spacebar selects the cell.
+			// For the up key the index is changed by negative the x dimension.
+
+			var keyIncrementMap = {
+				UP_ARROW: -this._xDim,
+				// The down key the index is increase by the x dimension.
+				DOWN_ARROW: this._xDim,
+				// Right and left move the index by 1.
+				RIGHT_ARROW: this.isLeftToRight() ? 1 : -1,
+				LEFT_ARROW: this.isLeftToRight() ? -1 : 1
+			};
+			for(var key in keyIncrementMap){
+				this.own(
+					typematic.addKeyListener(
+						this.domNode,
+						{keyCode: keys[key], ctrlKey: false, altKey: false, shiftKey: false},
+						this,
+						function(){
+							var increment = keyIncrementMap[key];
+							return function(count){
+								this._navigateByKey(increment, count);
+							};
+						}(),
+						this.timeoutChangeRate,
+						this.defaultTimeout
+					)
+				);
+			}
+		},
+
+		postCreate: function(){
+			this.inherited(arguments);
+
+			// Set initial navigable node.
+			var cell = this._getValueCell();
+			if (cell) {
+				this._setCurrent(cell.node);
+			}
```

Comment:
Should likely default to the first cell, `this._cells[0]`, like in the original behavior, when `value` is not defined or cannot be matched with the existing palette.
```js
var cell = this._getValueCell() || this._cells[0];
```

Reply:
addressed